ecstask/ecs_task.go

```diff
@@ -397,44 +402,121 @@ func removeAFromB(a, b []string) []string {
 	return diff
 }
 
-func deregisterTaskDefinitions(svc *ecs.ECS, taskDefinitionArns []string, parallel int) {
-	arnsChan := make(chan string, len(taskDefinitionArns))
+// Job carries information through a Job channel.
+type Job struct {
+	Arn string
+}
+
+// Result carries information through a Result channel.
+type Result struct {
+	Arn string
+	Err error
+}
+
+func deregisterTaskDefinitions(svc *ecs.ECS, taskDefinitionArns []string, parallel int, debug bool) {
```

Comment:
the logic in this function took me a bit to figure out, and I suspect it's partially due to order, (function) size, and naming. Call me crazy, but i think it would read more cleanly if `worker` and `dispatcher` were separate functions. Why? Most of this code is very procedural until we get here, and then suddenly we're slightly out-of-order via instantiating `worker` and `dispatcher`, not to mention they don't seem directly related to the stated goal of `deregisterTaskDefinitions`.

Reply:
That's a fair feeling, and I'd feel good about making them their own functions. Might just be an artifact of organization based on what helped me conceptualize and compartmentalize instead of organization based on readability.

Reply:
✅